src/pages/_app.tsx

```diff
@@ -5,16 +5,20 @@ import { SessionProvider } from "next-auth/react";
 
 import "../styles/globals.css";
 import NavRail from "../components/NavRail";
+import { useMediaQuery } from "../utils/useMediaQuery";
+import MobileNavBar from "../components/MobileNavBar";
 
 const MyApp: AppType<{ session: Session | null }> = ({
   Component,
   pageProps: { session, ...pageProps },
 }) => {
+  const isMd = useMediaQuery("md");
+
   return (
     <SessionProvider session={session}>
-      <div className="flex h-max bg-surface-background">
-        <NavRail />
-        <main className="min-h-screen w-full px-8 pt-8">
+      <div className="h-max w-screen bg-surface-background md:flex">
+        {isMd ? <NavRail /> : <MobileNavBar />}
```

Comment:
Can you put this in tailwind instead using my suggestion earlier? Wrap both these in a div with those tailwind classNames and that'll handle conditional rendering